index.js

```diff
@@ -316,6 +316,10 @@ function HttpRequest(options) {
 			}
 		};
 	}
+
+	if ('onerror' in xhr && !xhr.onreadystatechange) {
+		xhr.onerror = onLoad;
```

Comment:
You could move this straight to the if-clause above, right? `if ('onload' in xhr && 'onerror' in xhr)`? Else we'll just always use readyState.